wordpress-mobile/WordPress-iOS
# Integrate Gravatar Quick Editor

Fixes #

Integrates the Gravatar Quick Editor into Jetpack.
Changes are behind the remote feature flag: `gravatar_quick_editor`

<img src="https://github.com/user-attachments/assets/916ae964-2591-4022-9cf6-f9cd9a37691a" width=300/>

https://github.com/user-attachments/assets/a5baef59-fe18-4ff5-a5ec-211a156bbf21

https://github.com/user-attachments/assets/09521681-f8cc-45f5-8c52-ec4cc901dd5a

**Technical Details**

The new structure depends on the Quick Editor's update callback and fires a notification `GravatarQEAvatarUpdateNotification`. The UI components that render the avatar subscribe to that notification. Once received, they fetch the latest version of the avatar from the BE. To achieve this we introduce a way to `forceRefresh` the avatar, which wasn't present before. To achieve `forceRefresh` we add a cache buster query parameter otherwise the backend cache gets in the way and the latest version of the avatar can't be fetched.

**To test:**

**Enable the feature flag** "Gravatar Quick Editor" in `Me > App Settings > Debug > Feature Flags` (It's enabled by default in DEBUG config)
Log out from the Jetpack app
Log in with a new email (Highly recommend getting a temp email via https://tempmail.so/us , you should go to your inbox in your mobile device since the login depends on a magic link)
Use the email magic link to continue to the app
Observe: You landed on the signup epilogue, the avatar placeholder has a + on it and tapping on the avatar opens the Quick Editor.
Upload & change the avatar via the Quick Editor
Observe: The avatar on the signup page also gets updated
Close the Quick Editor
Continue to the app

Go to `Me > My Profile`
Tap either on the avatar or on the "Update profile photo" button
Observe: Quick Editor opens
Upload & change the avatar via the Quick Editor
Observe: The avatars on the main app gets updated as well (the one on the profile header, and the "Me" tab bar item at the bottom)

**Appearance Test**
Go to Me > App Settings > Appearance
Switch the appearance
Go to `Me > My Profile`
Tap either on the avatar or on the "Update profile photo" button
Observe: Quick Editor opens and the appearance matched with the app.

**Disable the feature flag** "Gravatar Quick Editor" in `Me > App Settings > Debug > Feature Flags` 

Repeat the tests ☝️
This time instead of the Gravatar Quick Editor you should observe that the system photo picker appears and the selected image is uploaded & set as the avatar.

## Regression Notes
1. Potential unintended areas of impact
The existing upload flow

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Tested with the feature flag off

3. What automated tests I added (or what prevented me from doing so)
x

PR submission checklist:


Testing checklist:
